src/NewsService.Business/GetNewsCommand.cs

```diff
@@ -15,29 +28,116 @@ public class GetNewsCommand : IGetNewsCommand
+    {
+      try
+      {
+        IOperationResult<IGetUsersDataResponse> response = _rcGetUsers.GetResponse<IOperationResult<IGetUsersDataResponse>>(
+          IGetUsersDataRequest.CreateObj(new List<Guid> { userId })).Result.Message;
+
+        if (response.IsSuccess)
+        {
+          return response.Body.UsersData.FirstOrDefault();
+        }
+
+        _logger.LogWarning(
+          $"Can not get author. Reason:{Environment.NewLine}{string.Join('\n', response.Errors)}.");
+      }
+      catch (Exception errorMessage)
+      {
+        _logger.LogError(errorMessage, "Cannot get author now. Please try again later.");
+      }
+
+      errors.Add($"Can not get author info for authorId '{userId}'. Please try again later.");
+
+      return null;
+    }
+
+    private DepartmentInfo GetDepartment(Guid? departmentId, List<string> errors)
+    {
+      if (departmentId == null)
+      {
+        return null;
+      }
+
+      try
+      {
+        IOperationResult<IGetDepartmentsResponse> departmentResponse =
+          _rcGetDepartments.GetResponse<IOperationResult<IGetDepartmentsResponse>>(
+            IGetDepartmentsRequest.CreateObj(new List<Guid> { departmentId.Value }))
+          .Result.Message;
+
+        if (departmentResponse.IsSuccess)
+        {
+          return _departmentInfoMapper.Map(departmentResponse.Body.Departments.FirstOrDefault());
+        }
+
+        _logger.LogWarning(
+          $"Can not get department. Reason:{Environment.NewLine}{string.Join('\n', departmentResponse.Errors)}.");
+      }
+      catch (Exception errorMesssage)
+      {
+        _logger.LogError(errorMesssage, "Exception on get department request.");
+      }
+
+      errors.Add($"Can not get department info for DepartmentId '{departmentId}'. Please try again later.");
+
+      return null;
+    }
 
     public GetNewsCommand(
       INewsRepository repository,
       INewsResponseMapper mapper,
-      IHttpContextAccessor httpContextAccessor)
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<IGetDepartmentsRequest> rcGetDepartments,
+      IRequestClient<IGetUsersDataRequest> rcGetUsers,
+      IDepartmentInfoMapper departmentMapper,
+      ILogger<GetNewsCommand> logger)
     {
       _repository = repository;
       _mapper = mapper;
       _httpContextAccessor = httpContextAccessor;
+      _rcGetDepartments = rcGetDepartments;
+      _rcGetUsers = rcGetUsers;
+      _departmentInfoMapper = departmentMapper;
+      _logger = logger;
     }
 
     public OperationResultResponse<NewsResponse> Execute(Guid newsId)
     {
       OperationResultResponse<NewsResponse> response = new();
 
-      response.Body = _mapper.Map(_repository.Get(newsId));
+      DbNews dbNews = _repository.Get(newsId);
+      if (dbNews == null)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+
+        response.Errors = new() { "News can't must be null" };
```

Comment:
the strange message, how can it help the user?

Reply:
done